hr_holidays_public/models/resource_calendar.py

```diff
@@ -15,20 +15,42 @@ class ResourceCalendar(models.Model):
     def _attendance_intervals_batch_exclude_public_holidays(
         self, start_dt, end_dt, intervals, resources, tz
     ):
-        list_by_dates = (
-            self.env["hr.holidays.public"]
-            .get_holidays_list(
-                start_dt=start_dt.date(),
-                end_dt=end_dt.date(),
-                employee_id=self.env.context.get("employee_id", False),
+        holidays_by_country = {
+            False: set(
+                self.env["hr.holidays.public"]
+                .get_holidays_list(
+                    start_dt=start_dt.date(),
+                    end_dt=end_dt.date(),
+                    employee_id=self.env.context.get("employee_id", False),
+                )
```

Comment:
In case `self.env.context.get("employee_id")` is set, shouldn't the this dict key start with the employee's country.id rather than `False` ?